tests/run/colltest4/CollectionStrawMan4_1.scala

```diff
@@ -217,7 +217,7 @@ object CollectionStrawMan4 {
     def fromIterable[B](coll: Iterable[B]): ListBuffer[B] = coll match {
       case pd @ View.Partitioned(partition: View.Partition[B]) =>
         partition.distribute(new ListBuffer[B]())
-        pd.forced.get.asInstanceOf[ListBuffer[B]]
+        new ListBuffer[B] ++= pd.forced.get.asInstanceOf[List[B]]
```

Comment:
The code looks correct to me. The cast seems to be a left-over from a
previous version.

On Fri, Jun 3, 2016 at 2:45 PM, Guillaume Martres <notifications@github.com>
wrote:

> In tests/run/colltest4/CollectionStrawMan4_1.scala
> <https://github.com/lampepfl/dotty/pull/1289#discussion_r65700740>:
>
> > @@ -217,7 +217,7 @@ object CollectionStrawMan4 {
> >      def fromIterable[B](coll: Iterable[B]): ListBuffer[B] = coll match {
> >        case pd @ View.Partitioned(partition: View.Partition[B]) =>
> >          partition.distribute(new ListBuffer[B]())
> > -        pd.forced.get.asInstanceOf[ListBuffer[B]]
> > +        new ListBuffer[B] ++= pd.forced.get.asInstanceOf[List[B]]
>
> You don't even need the cast, pd.forced.get works, but it would be
> interesting to know if this is what @odersky <https://github.com/odersky>
> intented or if there's a design flaw somewhere. Also please squash your
> commits related to this into one commit
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/lampepfl/dotty/pull/1289/files/61ce7bb9be6a74d19b246e621f8fbc02581684e5..8f3b4bf6cc2927dc2b1d9e9515dd0451b5359233#r65700740>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAwlVoqkdv3GnuHJWa4SDmvsTQq6NGcfks5qICHWgaJpZM4IrGUX>
> .
>



-- 
Prof. Martin Odersky
LAMP/IC, EPFL